### ISSUE TITLE
About the corresponding code of the paper

### ISSUE THREAD

**Issue by Grople** [nstucki/Betti-matching#6]
Hello stucki, Your work is fantastick, so I read your paper and code <<Topologically Faithful Image Segmentation via Induced Matching of Persistence Barcodes>> more than a week, but I very hard to follow your idea, especailly the connections between the paper and code, due to my ability is not enough. If providing more annotatation in code is better.
The following question makes me confused:
1. Union find. I can't find the detail to explain the logic of code in paper. 
![de9beaeb8dadff299a54d9147a733912](https://github.com/user-attachments/assets/38c8ba7b-7309-4157-b0af-1e27f6399cd3)
![bcd99faca24d66a02aa6653aa708fabf](https://github.com/user-attachments/assets/bf014bfb-a478-4220-a32f-0d4860651ea5)
![4eb2dc97b6111820666b2b3cf66c7c52](https://github.com/user-attachments/assets/54483139-6ccd-4e43-a286-f6060f5a8521)

2. Find function. In the code, I see the function "find" exists in cubical complex, image comlex, and induce mathc. Could you mind explain the difference of significance among these three one. 

3. Induce match. It is very confused me,  how to impelement the this function in picture. I find the code and  pseduecode，the critical problem is the difference between ImagePersistence and CubicalPersistency these two class. 

![image](https://github.com/user-attachments/assets/05ec68b3-51a8-42fe-a9c3-f9714b23a14d)

That's all, look forward to your answer.

**Comment by nstucki** [comment]
Hello Grople,

I will try to answer your questions:

1. The UnionFind data-structure (see https://en.wikipedia.org/wiki/Disjoint-set_data_structure) is used to compute persistent homology in dimension 0. It is not referring to the union of the refined barcodes in the paper.

2. Unfortunately I don't know which "find" function you are referring to. 

3. The induced matchings are computed by the method "match" of the class "InducedMatching".

Best,
Nico

**Comment by Grople** [comment]
thanks for your answer, please wait for several days to consider.

**Comment by Grople** [comment]
hello nstucki, there is confused in the paper ,  How do you determin whether are matched between two persistence barcode

**Comment by nstucki** [comment]
Hi Grople,

I am not sure if I understand your question correctly. But if you are asking how to determine which cycles are matched, then the answer is basically described in the paper. 
To circumscribe it briefly, you determine the induced matchings with respect to the inclusions into the comparison image by computing the image barcodes and then determine their composition.

Best,
Nico
